integration-tests/tls_test.go

```diff
@@ -1056,16 +1057,36 @@ func testProxyClientTls(t *testing.T, ccmSetup *setup.CcmTestSetup,
 
 	logMessages := buffer.String()
 
+	warnCheckFail := false
+	warnCheckDone := false
 	for _, errWarnExpected := range proxyTlsConfig.errWarningsExpected {
-		require.True(t, strings.Contains(logMessages, errWarnExpected), "%v not found", errWarnExpected)
+		warnCheckDone = true
+		if !strings.Contains(logMessages, errWarnExpected) {
+			t.Logf("%v not found in %v", errWarnExpected, logMessages)
+			warnCheckFail = true
+		}
 	}
 
 	if proxyTlsConfig.errExpected {
 		require.NotNil(t, err, "Did not get expected error %s", proxyTlsConfig.errMsgExpected)
+		errCheckFail := false
+		errCheckDone := false
 		if proxyTlsConfig.errMsgExpected != "" {
-			require.True(t, strings.Contains(err.Error(), proxyTlsConfig.errMsgExpected), err.Error())
+			errCheckDone = true
+			if !strings.Contains(err.Error(), proxyTlsConfig.errMsgExpected) {
+				errCheckFail = true
+				t.Logf("%v not found in %v", err.Error(), proxyTlsConfig.errMsgExpected)
+			}
+		}
+		if errCheckDone && warnCheckDone {
+			require.True(t, !errCheckFail || !warnCheckFail) // only 1 check needs to pass in this scenario
```

Comment:
Nit: I think this would be clearer given the other requires are False too:

`require.False(t, errCheckFail && warnCheckFail)`

Reply:
I actually set this to `require.True` because to me it felt easier to understand but since I added a comment anyway I can use `require.False` to be consistent with the others

Reply:
done